app/components/message_filter.py

```diff
@@ -45,15 +51,38 @@ async def check_message_filters(message: discord.Message) -> bool:
     for msg_filter in MESSAGE_FILTERS:
         if message.channel.id != msg_filter.channel_id or msg_filter.filter(message):
             continue
+
         await message.delete()
+
+        # Don't DM the user if it's a system message
+        # (e.g. "@user started a thread")
         if message.type not in REGULAR_MESSAGE_TYPES:
             continue
+
         assert isinstance(message.channel, discord.TextChannel)
-        await try_dm(
-            message.author,
-            MESSAGE_DELETION_TEMPLATE.format(
-                message.channel.mention, *msg_filter.template_fillers
-            ),
+
+        notification = MESSAGE_DELETION_TEMPLATE.format(
+            message.channel.mention, *msg_filter.template_fillers
         )
+        if content_size := len(message.content):
+            notification += MESSAGE_CONTENT_NOTICE
+        await try_dm(message.author, notification, silent=content_size > 0)
+
+        if content_size > 2000:
+            # The user has Nitro but the bot doesn't,
+            # so we're packing it into a file
+            await try_dm(
+                message.author,
+                "",
+                file=discord.File(
+                    BytesIO(message.content.encode()), filename="content.md"
+                ),
+            )
+        elif content_size > 0:
+            await try_dm(message.author, message.content)
+
+        if 0 < content_size <= 2000:
+            await try_dm(message.author, COPY_TEXT_HINT, silent=True)
+
```

Comment:
Can the logic for long message goes into `try_dm` itself? Feels like we have to do that every time `try_dm` is used anyway

Reply:
Other than here, this kind of check is only ever used in the moving messages feature where `try_dm` isn't used. But I agree this can probably be abstracted away for reuse 
and plugged into `try_dm` as a built-in safeguard 👍

Reply:
Alrighty. I felt like bringing this up since I have seen similar snippets on 2000 in some other pre. Looks like you already have a plan though :D